test_runner/src/test/kotlin/ftl/util/UtilsTest.kt

```diff
@@ -180,64 +176,76 @@ class UtilsTest {
     @Test
     fun `should terminate process with exit code 10 if FailedMatrix exception is thrown`() {
         // given
-        val block = { throw FailedMatrixError(listOf(testMatrix1, testMatrix2)) }
+        val exception = FailedMatrixError(listOf(testMatrix1, testMatrix2))
 
-        // will
-        exit.expectSystemExitWithStatus(NOT_PASSED)
-
-        // when
-        withGlobalExceptionHandling(block)
+        // when - then
+        withGlobalExceptionHandling(
+            { throw exception },
+            {
+                assertThat(it).isEqualTo(NOT_PASSED)
+            }
+        )
     }
 
     @Test
     fun `should terminate process with exit code 2 if YmlValidationError is thrown`() {
-        exit.expectSystemExitWithStatus(CONFIGURATION_FAIL)
-        val block = { throw YmlValidationError() }
-        withGlobalExceptionHandling(block)
+        // given
+        val exception = YmlValidationError()
+
+        // when - then
+        withGlobalExceptionHandling(
+            { throw exception },
+            { exitCode ->
+                assertThat(exitCode).isEqualTo(CONFIGURATION_FAIL)
+            }
+        )
     }
 
     @Test
     fun `should terminate process with exit code 1 and cancel running matrices if FlankTimeoutError is thrown`() {
         // given
         mockkStatic("ftl.run.CancelLastRunKt")
         coEvery { cancelMatrices(any(), any()) } just runs
-        exit.expectSystemExitWithStatus(1)
-        val block = { throw FlankTimeoutError(mapOf("anyMatrix" to mockk(relaxed = true)), "anyProject") }
-        // when
-        withGlobalExceptionHandling(block)
-        // then
-        coVerify(exactly = 1) { cancelMatrices(any(), any()) }
+        val exception = FlankTimeoutError(mapOf("anyMatrix" to mockk(relaxed = true)), "anyProject")
+
+        // when-then
+        withGlobalExceptionHandling(
+            { throw exception },
```

Comment:
hmmm shall we enable ktlint for tests. I am not liking this styling. ? WDYT?

Reply:
I will do it in separate PR

Reply:
@Sloox what is wrong with this styling? IMO it's better than
```
{
    throw exception
},
```
cause it's shorter